src/main/java/com/github/cameltooling/lsp/internal/codeactions/CamelKafkaConnectorCamelURLRefactor.java

```diff
@@ -0,0 +1,134 @@
+import com.github.cameltooling.lsp.internal.parser.CamelKafkaUtil;
+import com.github.cameltooling.lsp.internal.parser.ParserFileHelper;
+import com.github.cameltooling.lsp.internal.parser.ParserFileHelperFactory;
+import com.github.cameltooling.lsp.internal.parser.ParserFileHelperUtil;
+
+public class CamelKafkaConnectorCamelURLRefactor {
+
+	public static final String CONVERT_TO_LIST_OF_PROPERTIES_NOTATION = "Convert to list of properties notation.";
+	
+	private CamelTextDocumentService camelTextDocumentService;
+
+	public CamelKafkaConnectorCamelURLRefactor(CamelTextDocumentService camelTextDocumentService) {
+		this.camelTextDocumentService = camelTextDocumentService;
+	}
+
+	public Collection<Either<Command, CodeAction>> getCodeActions(CodeActionParams params) {
+		String uri = params.getTextDocument().getUri();
+		if (uri.endsWith(".properties")) {
+			TextDocumentItem openedDocument = camelTextDocumentService.getOpenedDocument(uri);
+			int startLine = params.getRange().getStart().getLine();
+			int endLine = params.getRange().getEnd().getLine();
+			if (startLine == endLine) {
+				String line = new ParserFileHelperUtil().getLine(openedDocument, startLine);
+				int equalIndex = line.indexOf('=');
+				if (equalIndex != -1) {
+					String propertyKey = line.substring(0, equalIndex);
+					if (new CamelKafkaUtil().isCamelURIForKafka(propertyKey)) {
+						ParserFileHelper parserFileHelper = new ParserFileHelperFactory().getCorrespondingParserFileHelper(openedDocument, startLine);
+						if (parserFileHelper != null) {
+							String camelComponentUri = parserFileHelper.getCamelComponentUri(openedDocument, new Position(startLine, equalIndex + 1));
+							if (camelComponentUri != null) {
+								TextEdit textEdit = createTextEdit(openedDocument, startLine, line, equalIndex, parserFileHelper, camelComponentUri);
+								return createCodeAction(uri, textEdit);
+							}
+						}
+					}
+				}
+			}
+		}
+		return Collections.emptyList();
+	}
+
+	private Collection<Either<Command, CodeAction>> createCodeAction(String uri, TextEdit textEdit) {
+		List<Either<Command, CodeAction>> codeActions = new ArrayList<>();
+		Map<String, List<TextEdit>> changes = new HashMap<>();
+		changes.put(uri, Arrays.asList(textEdit));
+		CodeAction codeAction = new CodeAction(CONVERT_TO_LIST_OF_PROPERTIES_NOTATION);
+		codeAction.setKind(CodeActionKind.Refactor);
+		codeAction.setEdit(new WorkspaceEdit(changes));
+		codeActions.add(Either.forRight(codeAction));
+		return codeActions;
+	}
+
+	private TextEdit createTextEdit(TextDocumentItem openedDocument, int startLine, String line, int equalIndex, ParserFileHelper parserFileHelper, String camelComponentUri) {
+		CamelURIInstance camelURIInstance = parserFileHelper.createCamelURIInstance(openedDocument, new Position(startLine, equalIndex + 1), camelComponentUri);
+		String sinkOrSource = line.startsWith(CamelKafkaUtil.CAMEL_SINK_URL) ? "sink" : "source";
```

Comment:
`sink` and `source` are just used once in this class but it might be better to extract them into constants too. Wdyt?